File: packages/client-android/src/main/kotlin/dev/poppy/android/Tokens.kt
/*
 * Token enums for the v0.1 wire format. Mirrors `packages/schema/src/types.ts`.
 *
 * Wire-format identifiers are lowercase strings; we map them via @SerialName
 * on each enum value. Renderers consume the enum, the JSON layer never escapes.
 *
 * The token *values* (e.g. md = 16.dp) live in [PoppyTheme]; this file only
 * declares the names.
 */

package dev.poppy.android

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class Spacing {
    @SerialName("none") NONE,
    @SerialName("xs") XS,
    @SerialName("sm") SM,
    @SerialName("md") MD,
    @SerialName("lg") LG,
    @SerialName("xl") XL,
}

@Serializable
enum class Size {
    @SerialName("xs") XS,
    @SerialName("sm") SM,
    @SerialName("md") MD,
    @SerialName("lg") LG,
    @SerialName("xl") XL,
}

@Serializable
enum class Color {
    @SerialName("default") DEFAULT,
    @SerialName("primary") PRIMARY,
    @SerialName("secondary") SECONDARY,
    @SerialName("danger") DANGER,
    @SerialName("success") SUCCESS,
}

@Serializable
enum class Weight {
    @SerialName("regular") REGULAR,
    @SerialName("medium") MEDIUM,
    @SerialName("bold") BOLD,
}

@Serializable
enum class Alignment {
    @SerialName("start") START,
    @SerialName("center") CENTER,
    @SerialName("end") END,
    @SerialName("stretch") STRETCH,
}

@Serializable
enum class Fit {
    @SerialName("contain") CONTAIN,
    @SerialName("cover") COVER,
    @SerialName("fill") FILL,
}

@Serializable
enum class Axis {
    @SerialName("horizontal") HORIZONTAL,
    @SerialName("vertical") VERTICAL,
}

File: packages/client-ios/Sources/Poppy/Tokens.swift
// Token enums for Poppy v0.1.
//
// These mirror packages/schema/src/types.ts and the JSON Schema files in
// packages/schema/schemas/tokens/. The conformance corpus catches drift.
//
// Each token's raw value matches the wire-format string exactly so Codable's
// synthesised init(from:) handles decoding directly.

import Foundation

public enum Spacing: String, Codable, Sendable, Equatable, Hashable, CaseIterable {
    case none
    case xs
    case sm
    case md
    case lg
    case xl
}

public enum Size: String, Codable, Sendable, Equatable, Hashable, CaseIterable {
    case xs
    case sm
    case md
    case lg
    case xl
}

public enum PoppyColor: String, Codable, Sendable, Equatable, Hashable, CaseIterable {
    case `default`
    case primary
    case secondary
    case danger
    case success
}

public enum Weight: String, Codable, Sendable, Equatable, Hashable, CaseIterable {
    case regular
    case medium
    case bold
}

public enum PoppyAlignment: String, Codable, Sendable, Equatable, Hashable, CaseIterable {
    case start
    case center
    case end
    case stretch
}

public enum Fit: String, Codable, Sendable, Equatable, Hashable, CaseIterable {
    case contain
    case cover
    case fill
}

public enum Axis: String, Codable, Sendable, Equatable, Hashable, CaseIterable {
    case horizontal
    case vertical
}

/// Hint passed to `PoppyHost.isUrlAllowed(_:context:)` so a host can apply
/// different allowlists per use site (image vs. future media types).
public enum ImageContext: String, Sendable, Equatable, Hashable {
    case image
}
